docs/source/traits_api_reference/trait_handlers.rst

```diff
@@ -27,15 +23,33 @@ Classes
 
 .. autoclass:: TraitCompound
 
-.. autoclass:: TraitTuple
-
-.. autoclass:: TraitList
-
-.. autoclass:: TraitDict
-
 Private Functions
 -----------------
 
 .. autofunction:: traits.trait_handlers._undefined_get
 
 .. autofunction:: traits.trait_handlers._undefined_set
+
+Deprecated Handlers
+-------------------
+
+.. deprecated:: 6.0.0
+
+The following :class:`~.TraitHandler` classes and instances are deprecated,
+and may be removed in a future version of Traits.
+
+.. autodata:: ThisClass
+
+.. autodata:: TraitDict
+
+.. autodata:: TraitEnum
+
+.. autodata:: TraitFunction
```

Comment:
It looks as though `TraitFunction` is still in the original list (above), too; it should be removed from there.